metadb/component.go

```diff
@@ -0,0 +1,159 @@
+package metadb
```

Comment:
Should this be part of store package which is the package handling our metadata?

Reply:
I prefer to leave it in package `component`, stand with @tianzhou .

Reply:
What's the reasons (pros/cons) of having a separate component package?

I'm trying to minimize the top level packages. Also the logic is the initial step of creating a store connection so it makes more sense to keep them together.

Reply:
SG. At a high level, embed Postgres more like a component, I will think that MySQL is a component too if we embed MySQL. But in logical view, this should be a part of the store package. I will move it to the store package.

Reply:
Yeah, if we really feel awkward about PostgreSQL installation, that particular installation logic can be part of root.go start() instead of server.New(). But that's really optional and can be a follow-up PR.

Reply:
I will fix it in a follow-up PR. To make it prettier is really hard work.
In other words, if we let install Postgres be a part of start(), the test needs to be responsible for installing Postgres by itself. I guess we can test external Postgres easier.